store/boltstore/bolt_store_test.go

```diff
@@ -414,3 +434,212 @@ func TestBoltStore_SetUint64_GetUint64(t *testing.T) {
 		t.Fatalf("bad: %v", val)
 	}
 }
+
+func TestBoltStore_Snapshot_Create(t *testing.T) {
+	store := testBoltStore(t)
+	defer store.Close()
+	defer os.RemoveAll(store.path)
+
+	// Check no snapshots
+	snaps, err := store.List()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if len(snaps) != 0 {
+		t.Fatalf("did not expect any snapshots: %v", snaps)
+	}
+
+	// Create a new sink
+	var configuration raft.Configuration
+	configuration.Servers = append(configuration.Servers, raft.Server{
+		Suffrage: raft.Voter,
+		ID:       raft.ServerID("my id"),
+		Address:  raft.ServerAddress("over here"),
+	})
+	_, trans := raft.NewInmemTransport(raft.NewInmemAddr())
+	sink, err := store.Create(SnapshotVersionMax, 10, 3, configuration, 2, trans)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	// The sink is not done, should not be in a list!
+	snaps, err = store.List()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if len(snaps) != 0 {
+		t.Fatalf("did not expect any snapshots: %v", snaps)
+	}
+
+	// Write to the sink
+	_, err = sink.Write([]byte("first\n"))
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	_, err = sink.Write([]byte("second\n"))
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	// Done!
+	err = sink.Close()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	// Should have a snapshot!
+	snaps, err = store.List()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if len(snaps) != 1 {
+		t.Fatalf("expect a snapshots: %v", snaps)
+	}
+
+	// Check the latest
+	latest := snaps[0]
+	if latest.Index != 10 {
+		t.Fatalf("bad snapshot: %v", *latest)
+	}
+	if latest.Term != 3 {
+		t.Fatalf("bad snapshot: %v", *latest)
+	}
+	if !reflect.DeepEqual(latest.Configuration, configuration) {
+		t.Fatalf("bad snapshot: %v", *latest)
+	}
+	if latest.ConfigurationIndex != 2 {
+		t.Fatalf("bad snapshot: %v", *latest)
+	}
+	if latest.Size != 13 {
+		t.Fatalf("bad snapshot: %v", *latest)
+	}
+
+	// Read the snapshot
+	_, r, err := store.Open(latest.ID)
```

Comment:
I think we should also check the snapshort metadata.

Reply:
I have checked it from line 500-515 by using ```List```. Because there's only one snapshot, ```List``` and ```Open``` actually return the same result. As long as the return error value of ```Open``` is ```nil```, we can guarantee that the snapshot of that ID is stored in bbolt and the check of it is the same of the check of ```List```.

Reply:
Good.